Add bootc transient package installation information (#3843)
theforeman/foreman-documentation: guides/common/modules/con_package-mode-and-image-mode-hosts.adoc

Image mode simplifies enterprise operations by enabling development, operations, and solution providers to use the same container-native tools and techniques to manage both applications and the underlying operating system.

In image mode, users can install and manage packages, but with limitations. 

> Job templates, depending on the implementation and category, will either enable the `bootc usr-overlay` or pass the `--transient` flag to ensure that packages are installed correctly.

Reviewer: It would be nice to rephrase this as an action to be performed by users. Written like this, I'm not entirely sure what myself, as a potential user, should do. Just guessing, but:

```suggestion
When creating job templates, enable the `bootc usr-overlay` command or pass the `--transient` option to ensure that packages are installed correctly.
```
One note: I'm suggesting `the [...] command` because that seemed to be missing and s/flag/option because by convention, we don't use "flag" here.

Author: This line was meant to be more informational noting that our job templates will either enable the usr-overlay or pass the transient flag into dnf depending on if the host is an image mode host or not.

Author: So, there's no action that the user has to take besides understanding that their packages will be installed transiently (which, well, we do have warnings in the UI about it, but it felt nice putting it here too).

Reviewer: Okay, can you please think of a way to avoid "will enable" (to avoid future tense here) and add the missing qualifying word ("the `bootc usr-overlay` something")? That's just to comply with our style guide.

Author: I've updated it now.

Any packages installed on image mode hosts are temporary and disappear after a reboot. 
Because of this, installing packages is generally discouraged, except for debugging purposes or, in some cases, when applying critical security patches.

> If new packages are needed on an image mode host, consider building a new container image for the host with the necessary packages built-in.

Author: ```suggestion
If new packages are needed on an image mode host, consider building a new container image for the host with the necessary packages included.
```
Note to self: saying building/built in the same sentence is a little awkward.

> If new packages are needed on an image mode host, consider building a new container image for the host with the necessary packages built-in.

Reviewer: What would you think about this? I'm trying to remove the need for passive voice and perhaps also simplify the sentence structure a little bit.
```suggestion
If you need new packages on an image mode host, consider building a new container image that includes the necessary packages.
```
